pkg/provider/fake/fake.go

```diff
@@ -77,7 +86,6 @@ func (p *Provider) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecre
 		if data.Key != ref.Key || data.Version != ref.Version || data.ValueMap == nil {
 			continue
 		}
-		return convertMap(data.ValueMap), nil
```

Comment:
I think this return was removed, but we should keep it?

Reply:
Sorry, I've mistakenly included some changes I didn't intend to... I removed them!

Reply:
I don't feel GetSecretMap is correct since it does not seem to support the `Property` right now, but what do you think? 🤔

Reply:
In order to support property what we can do is save the convertMap as a variable, and then apply the same gjson logic to that variable, if Property is defined, of course. That should do the trick :)

Reply:
Thank you for your guidance! I've updated the PR again, so would you check if that aligns with your intention? 🙏 I'll squash the commits with singing once it LGTY 🙂